sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -0,0 +1,2191 @@
+        {
+            this.Uri = blobUri;
+            this.Pipeline = (options ?? new BlobClientOptions()).Build(authentication);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlobClient"/>
+        /// class.
+        /// </summary>
+        /// <param name="blobUri">
+        /// A <see cref="Uri"/> referencing the blob that includes the
+        /// name of the account, the name of the container, and the name of
+        /// the blob.
+        /// </param>
+        /// <param name="pipeline">
+        /// The transport pipeline used to send every request.
+        /// </param>
+        internal BlobBaseClient(Uri blobUri, HttpPipeline pipeline)
+        {
+            this.Uri = blobUri;
+            this.Pipeline = pipeline;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlobClient"/>
+        /// class with an identical <see cref="Uri"/> source but the specified
+        /// <paramref name="snapshot"/> timestamp.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/creating-a-snapshot-of-a-blob" />.
+        /// </summary>
+        /// <param name="snapshot">The snapshot identifier.</param>
+        /// <returns>A new <see cref="BlobClient"/> instance.</returns>
+        /// <remarks>
+        /// Pass null or empty string to remove the snapshot returning a URL
+        /// to the base blob.
+        /// </remarks>
+        public virtual BlobBaseClient WithSnapshot(string snapshot) => this.WithSnapshotImpl(snapshot);
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="BlobClient"/> class
+        /// with an identical <see cref="Uri"/> source but the specified
+        /// <paramref name="snapshot"/> timestamp.
+        /// </summary>
+        /// <param name="snapshot">The snapshot identifier.</param>
+        /// <returns>A new <see cref="BlobClient"/> instance.</returns>
+        protected virtual BlobBaseClient WithSnapshotImpl(string snapshot)
+        {
+            var builder = new BlobUriBuilder(this.Uri) { Snapshot = snapshot };
+            return new BlobBaseClient(builder.ToUri(), this.Pipeline);
+        }
+
+        /// <summary>
+        /// Creates a clone of this instance that references a version ID rather than the base blob.
+        /// </summary>
+        /// /// <remarks>
+        /// Pass null or empty string to remove the verion ID returning a URL to the base blob.
+        /// </remarks>
+        /// <param name="versionId">The version ID to use on this blob. An empty string or null indicates to use the base blob.</param>
+        /// <returns>The new <see cref="BlobClient"/> instance referencing the verionId.</returns>
+        //public virtual BlobBaseClient WithVersionId(string versionId) => this.WithVersionIdImpl(versionId);
```

Comment:
Status of the commented methods?

Reply:
These were commented out by Sean right before //build.